### PR TITLE
bugfix: fix batch sql getTableMeta error

### LINKED ISSUE #3244: batch sql getTableMeta error
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

image

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

### Ⅰ. Describe what this PR did

bugfix: fix batch sql getTableMeta error

### Ⅱ. Does this pull request fix one issue?
fix #3244

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java [hunk 1]
```diff
@@ -98,6 +105,28 @@ protected T executeAutoCommitFalse(Object[] args) throws Exception {
...
+                if (getTableMeta(sqlRecognizer.getTableName()).getPrimaryKeyOnlyName().size() > 1) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
+        Map<Object, Boolean> seen = new ConcurrentHashMap<>();
```
Comment (l81893521) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java:125]: Look like new HashMap that s enough.

**Review by jsbxyyx — APPROVED** [review]
LGTM

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**